hollow/src/main/java/com/netflix/hollow/core/memory/encoding/HashCodes.java

```diff
@@ -173,7 +173,7 @@ public static int hashTableSize(int numElements) {
             return 1;
         if (numElements < 3)
             return numElements * 2;
-        int sizeAfterLoadFactor = numElements * 10 / 7;
+        int sizeAfterLoadFactor = (int)((long)numElements * 10 / 7);
```

Comment:
I was going to suggest dividing by 7 first and them multiplying by 10, but this gives you a better spread.

We can still overflow if numElements * 10 / 7 > Integer.MAX_VALUE - how about something like `Math.min(Integer.MAX_VALUE, numElements * 10L / 7)`?

Reply:
Also, TIL (can't wait for java8 with Hollow!) https://stackoverflow.com/questions/1590831/safely-casting-long-to-int-in-java

Reply:
Wouldn't `numElements * 10 / 7 > Integer.MAX_VALUE ` be totally out of Hollow's limits? I believe the max was ~500 million, right? if the limit for the number of ordinals is `Integer.MAX_VALUE` which would be the limit of any array in Java, then we might want a safer way to cast the ints? like the one proposed by @akhaku